File: Application/src/main/java/com/shareyourproxy/api/rx/event/ShareLinkEventCallback.java
package com.shareyourproxy.api.rx.event;

import android.os.Parcel;

import com.shareyourproxy.api.rx.command.eventcallback.EventCallback;

/**
 * A ShareLink message has been generated.
 */
public class ShareLinkEventCallback extends EventCallback {
    private final static java.lang.ClassLoader CL =
        ShareLinkEventCallback.class.getClassLoader();
    public static final Creator<ShareLinkEventCallback> CREATOR =
        new Creator<ShareLinkEventCallback>() {
            @Override
            public ShareLinkEventCallback createFromParcel(Parcel in) {
                return new ShareLinkEventCallback((String) in.readValue(CL));
            }

            @Override
            public ShareLinkEventCallback[] newArray(int size) {
                return new ShareLinkEventCallback[size];
            }
        };
    public final String message;

    public ShareLinkEventCallback(String message) {
        this.message = message;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(message);
    }
}

File: Application/src/main/kotlin/com/shareyourproxy/api/rx/command/eventcallback/ShareLinkEventCallback.kt
package com.shareyourproxy.api.rx.command.eventcallback

import android.os.Parcel
import android.os.Parcelable

/**
 * A ShareLink message has been generated.
 */
internal final class ShareLinkEventCallback(val message: String) : EventCallback() {

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeValue(message)
    }

    companion object {
        private val CL = ShareLinkEventCallback::class.java.classLoader
        val CREATOR: Parcelable.Creator<ShareLinkEventCallback> = object : Parcelable.Creator<ShareLinkEventCallback> {
            override fun createFromParcel(parcel: Parcel): ShareLinkEventCallback {
                return ShareLinkEventCallback(parcel.readValue(CL) as String)
            }

            override fun newArray(size: Int): Array<ShareLinkEventCallback?> {
                return arrayOfNulls(size)
            }
        }
    }
}
